src/readers/src/type2/type2_TRM.cpp

```diff
@@ -0,0 +1,71 @@
+#include "type2_TRM.h"
+
+#include <magic_enum.hpp>
+
+#include <any>
+#include <regex>
+
+using namespace magic_enum;
+
+namespace {
+void read_layers_info(std::istream& istream, Readers::Type2::LayerType& layer,
+                      int layers_num) {
+	if (layers_num > 0) {
+		std::string tmp;
+		istream >> layer.zc_.set_size(layers_num) >> tmp;
+		istream >> layer.vc_.set_size(layers_num) >> tmp;
+		istream >> layer.cc_.set_size(layers_num) >> tmp;
+		istream >> layer.rc_.set_size(layers_num) >> tmp;
```

Comment:
inside a class layer you have function `set_size` please use it and avoid code duplication